lib/foreman_tasks/tasks/export_tasks.rake

```diff
@@ -241,35 +241,42 @@ namespace :foreman_tasks do
       end
     end
 
-    def csv_export(export_filename, tasks)
+    def csv_export(export_filename, id_scope, task_scope)
       CSV.open(export_filename, 'wb') do |csv|
         csv << %w[id state type label result parent_task_id started_at ended_at duration]
-        tasks.find_each do |task|
-          with_error_handling(task) do
-            csv << [task.id, task.state, task.type, task.label, task.result,
-                    task.parent_task_id, task.started_at, task.ended_at, task.duration]
+        id_scope.pluck(:id).in_groups_of(1000, false).each do |ids|
```

Comment:
We don't really need anything that `in_groups_of` offers over `each_slice`, let's use that.